### PR TITLE
ad7134: Add multidevice synchronization feature

### PR DESCRIPTION
**Pull request by PIoandan**
## Pull Request Description

To achieve tight synchronization, the user must configure all the devices in slave mode and use the SPI to set the DIG_IF_RESET 
bit to reset the digital interface before the data capture. This DIG_IF_RESET command must be given to all the slaves 
simultaneously using one single SPI write command.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [ ] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

**Review by amiclaus — COMMENTED** [review]
hello, please sign the cla and fix the build issues in the first place.

File: projects/ad713x_fmcz/src/parameters.h [hunk 1]
```diff
@@ -52,7 +52,7 @@
 #define GPIO_DEVICE_ID			XPAR_PS7_GPIO_0_DEVICE_ID
 #define AD7134_DMA_BASEADDR		XPAR_AXI_AD7134_DMA_BASEADDR
 #define AD7134_SPI_ENGINE_BASEADDR	XPAR_DUAL_AD7134_DUAL_AD7134_AXI_REGMAP_BASEADDR
-#define AD713x_SPI_ENG_REF_CLK_FREQ_HZ	96000000
+#define AD713x_SPI_ENG_REF_CLK_FREQ_HZ	100000000
```
Comment (buha) [inline projects/ad713x_fmcz/src/parameters.h:55]: Can you please provide some context here ? There is this 96->100 MHz ref clock change and also SPI mode 3 to 0 change - do these mean that the project was not at all functional before ? Or did the FMCZ board change revisions and requires these changes ?

File: drivers/adc/ad713x/ad713x.c [hunk 2]
```diff
@@ -318,6 +318,24 @@ int32_t ad713x_mag_phase_clk_delay_chan(struct ad713x_dev *dev,
  * 							 CH3
  * @return 0 in case of success, -1 otherwise.
  */
+
+ int32_t ad713x_channel_sync(struct ad713x_dev *dev)
```
Comment (buha) [inline drivers/adc/ad713x/ad713x.c]: ad713x.c and ad713x.h contain the multisync feature, so bundle only these 2 in one commit.

The rest of the changes are separate things (like the SPI mode and ref clk etc.) and need to be presented as separate commits and documented in the commit description with a few sentences.

File: drivers/adc/ad713x/ad713x.c [hunk 3]
```diff
@@ -318,6 +318,24 @@ int32_t ad713x_mag_phase_clk_delay_chan(struct ad713x_dev *dev,
...
+{
+	int ret = 0;
+	ret = no_os_gpio_set_value(dev->gpio_cs_sync, true);
+		if (NO_OS_IS_ERR_VALUE(ret))
```
Comment (amiclaus) [inline drivers/adc/ad713x/ad713x.c]: please properly align this.

File: drivers/adc/ad713x/ad713x.c [hunk 4]
```diff
@@ -318,6 +318,24 @@ int32_t ad713x_mag_phase_clk_delay_chan(struct ad713x_dev *dev,
...
+			return -1;
+
+	ret = ad713x_spi_write_mask(dev, AD713X_REG_INTERFACE_CONFIG_B,
+						 AD713X_INT_CONFIG_B_DIG_IF_RST_MSK | AD713X_INT_CONFIG_B_SINGLE_INSTR_MSK,
+						 AD713X_INT_CONFIG_B_DIG_IF_RST_MSK | AD713X_INT_CONFIG_B_SINGLE_INSTR_MSK);
+
+	ret = no_os_gpio_set_value(dev->gpio_cs_sync, false);
+		if (NO_OS_IS_ERR_VALUE(ret))
```
Comment (amiclaus) [inline drivers/adc/ad713x/ad713x.c]: ditto.

File: drivers/adc/ad713x/ad713x.c [hunk 5]
```diff
@@ -408,6 +426,14 @@ static int32_t ad713x_init_gpio(struct ad713x_dev *dev,
 	if (NO_OS_IS_ERR_VALUE(ret))
 		return -1;
 
+	ret = no_os_gpio_get_optional(&dev->gpio_cs_sync, init_param->gpio_cs_sync);
+	if (NO_OS_IS_ERR_VALUE(ret))
+		return -1;
+
+	ret = no_os_gpio_direction_output(dev->gpio_cs_sync, false);
+		if (NO_OS_IS_ERR_VALUE(ret))
```
Comment (amiclaus) [inline drivers/adc/ad713x/ad713x.c]: ditto.

File: projects/ad713x_fmcz/src/ad713x_fmc.c [hunk 6]
```diff
@@ -244,9 +270,13 @@ int main()
 	if (ret != 0)
 		return -1;
 
-	ret = no_os_pwm_init(&axi_pwm, &axi_pwm_init);
-	if (ret != 0)
-		return ret;
+	ret = no_os_pwm_init(&axi_pwm, &axi_pwm_init_trigger);
+		if (ret != 0)
```
Comment (amiclaus) [inline projects/ad713x_fmcz/src/ad713x_fmc.c]: ditto.

File: projects/ad713x_fmcz/src/ad713x_fmc.c [hunk 7]
```diff
@@ -244,9 +270,13 @@ int main()
...
+			return ret;
+
+	ret = no_os_pwm_init(&axi_pwm, &axi_pwm_init_odr);
+		if (ret != 0)
```
Comment (amiclaus) [inline projects/ad713x_fmcz/src/ad713x_fmc.c]: ditto.

File: projects/ad713x_fmcz/src/ad713x_fmc.c [hunk 8]
```diff
@@ -348,6 +378,10 @@ int main()
 
 #endif /* IIO_SUPPORT */
 
+		ret = ad713x_channel_sync(ad713x_dev_1);
+		if (ret != 0)
```
Comment (amiclaus) [inline projects/ad713x_fmcz/src/ad713x_fmc.c]: ditto.

**Review by amiclaus — COMMENTED** [review]
please add a change log summarizing what you changed in the new version of the patches.

**Comment by PIoandan** [comment]
changes in v2:

Created separate commits for:
https://github.com/analogdevicesinc/no-OS/pull/2257/commits/9623300c3ff3e521f87496215a1fc10633554c01, https://github.com/analogdevicesinc/no-OS/pull/2257/commits/f47aba3c6fdd2412b01e844ed6d9ae5d7541e598 implementing the multidevice synchronization function,
https://github.com/analogdevicesinc/no-OS/pull/2257/commits/5832e9f2c37b158463b7b025991c18d0f3e58649 setting SPI mode,
https://github.com/analogdevicesinc/no-OS/pull/2257/commits/f9a2d18d48a1d6bc6bf49ae0b186a61c95ee9c97 setting the frequency of DCLK
https://github.com/analogdevicesinc/no-OS/pull/2257/commits/d7aed5d654652225d2549b598464aa6d919b0fd1 setting two PWM channels
Fixed the indentation issues.

File: drivers/adc/ad713x/ad713x.c [hunk 9]
```diff
@@ -318,6 +341,7 @@ int32_t ad713x_mag_phase_clk_delay_chan(struct ad713x_dev *dev,
  * 							 CH3
  * @return 0 in case of success, -1 otherwise.
  */
+
```
Comment (amiclaus) [inline drivers/adc/ad713x/ad713x.c:344]: nit: unrelated change.

File: projects/ad713x_fmcz/src/ad713x_fmc.c [hunk 10]
```diff
@@ -348,6 +361,10 @@ int main()
 
 #endif /* IIO_SUPPORT */
 
+	ret = ad713x_channel_sync(ad713x_dev_1);
+	if (ret != 0)
+		return -1;
```
Comment (amiclaus) [inline projects/ad713x_fmcz/src/ad713x_fmc.c:382]: return ret, so we are consistent with the rest of the code

File: projects/ad713x_fmcz/src/ad713x_fmc.c [hunk 11]
```diff
@@ -181,18 +181,30 @@ int main()
 	};
 
 	struct no_os_pwm_desc *axi_pwm;
-	struct axi_pwm_init_param axi_zed_pwm_init = {
+
```
Comment (amiclaus) [inline projects/ad713x_fmcz/src/ad713x_fmc.c:184]: drop this line.

**Review by amiclaus — COMMENTED** [review]
minor comments, rest looks good.

File: projects/ad713x_fmcz/src/ad713x_fmc.c (hunk 11 above)
Comment (PIoandan) [reply projects/ad713x_fmcz/src/ad713x_fmc.c]: Done.

File: projects/ad713x_fmcz/src/ad713x_fmc.c (hunk 10 above)
Comment (PIoandan) [reply projects/ad713x_fmcz/src/ad713x_fmc.c]: Done.

File: drivers/adc/ad713x/ad713x.c (hunk 9 above)
Comment (PIoandan) [reply drivers/adc/ad713x/ad713x.c]: Done.

**Review by buha — APPROVED** [review]

**Review by amiclaus — APPROVED** [review]
